### PR TITLE
fix(nuget): ignore build metadata

### LINKED ISSUE #7659: Wrong version for nuget package update
**What Renovate type, platform and version are you using?**

self-hosted with GitLab, but also on Azure DevOps. v23.73.0 

**Describe the bug**

One of the packages renovate has created a PR for recently was [Serilog.Exception](https://www.nuget.org/packages/Serilog.Exceptions/). As can be seen on the nuget page, in the "Version History" section the latest version is 5.7.0, however renovate creates a pull request changing the version to 5.7.0+build.146. Rider and Visual Studio do correctly show the latest version of the package is 5.7.0 and allow updating to that version, in fact, they don't even show the +build version as a version I can install.

The package version that is generated is still a valid version and it will build just fine, but it would be better if renovate could use the 5.7.0 version like Visual Studio and Rider show.

**Relevant debug logs**
```
DEBUG: package manifest (.nuspec) not found (repository=...)
       "registryUrl": "https://api.nuget.org/v3/index.json",
       "pkgName": "Serilog.Exceptions",
       "pkgVersion": "5.7.0+build.146"
DEBUG: packageFiles with updates (repository=...)
       "config": {
         "nuget": [
           {
             "packageFile": "xxx.csproj",
             "manager": "nuget",
             "deps": [
               {
                 "datasource": "nuget",
                 "depType": "nuget",
                 "depName": "Serilog.Exceptions",
                 "currentValue": "5.6.0",
                 "depIndex": 0,
                 "updates": [
                   {
                     "fromVersion": "5.6.0",
                     "toVersion": "5.7.0+build.146",
                     "newValue": "5.7.0+build.146",
                     "newMajor": 5,
                     "newMinor": 7,
                     "updateType": "minor",
                     "isSingleVersion": true,
                     "releaseTimestamp": "2020-11-03T15:49:51.55+00:00"
                   }
                 ],
                 "warnings": [],
                 "fixedVersion": "5.6.0"
               }
             ]
           }
         ]
       }

```

**To Reproduce**
create a new .net project and reference Serilog.Exceptions 5.6.0. When running renovate it will create a PR for 5.7.0+build.146

**Additional context**

### PR DESCRIPTION
**Pull request by viceice**
## Changes:

Remove the semver v2 build metadata from version.

## Context:

closes #7659

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] Unit tests + ran on a real repository

### CODE REVIEW

**Comment by viceice** [comment]
@fgreinacher What do you think about this change?

**Comment by JamieMagee** [comment]
Looks fine to me. [Here's this regex tested against the cases from semver.org](https://regex101.com/r/Ly7O1x/686)

File: lib/datasource/nuget/common.ts [hunk 1]
```diff
@@ -1 +1,7 @@
 export const id = 'nuget';
+
+const buildMetaRe = /\+.+$/g;
+
+export function replaceBuildMeta(version: string): string {
```
Comment (JamieMagee) [inline lib/datasource/nuget/common.ts:5]: nit: I'd name this `removeBuildMeta`, but I can live with it either way.

**Review by JamieMagee — DISMISSED** [review]

**Review by fgreinacher — COMMENTED** [review]
Technically I don't see any problems with this approach. But I'm wondering whether it might make more sense to handle this case at a higher level, e.g. in the versioning logic?

**Comment by rarkins** [comment]
Based on what we know, handling it in datasource seems right. Otherwise we'd have a problem once we try to insert the full version string into package files later. 

Question is: are there any package files that *do* want the full version string including build number?

**Comment by viceice** [comment]
> Technically I don't see any problems with this approach. But I'm wondering whether it might make more sense to handle this case at a higher level, e.g. in the versioning logic?

For this we would need to change the versionig api, so the versioning always decides the new value to write. This would require a lot more work, so i suggest my solution as short term fix and open a new issue for extending versioning.


> Based on what we know, handling it in datasource seems right. Otherwise we'd have a problem once we try to insert the full version string into package files later.
> 
> Question is: are there any package files that _do_ want the full version string including build number?

Nothing I'm aware of.

**Review by rarkins — APPROVED** [review]

**Comment by fgreinacher** [comment]
Got it, still feels a bit weird to remove this metadata here, but of course let's go for the low-effort solution first 👍

**Pull request merged**